BigDecimal arb can return edgecases outside min max limits #2834

ciffer opened the pull request:
Edgecases for Arb.bigDecimal(min, max) are now always contrained to be in range [min, max)
  Min is now added as edgecase for Arb.bigDecimal(min, max)

Commit:
BigDecimal arb can return edgecases outside min max limits #2834
  Edgecases for Arb.bigDecimal(min, max) are now always contrained to be in range [min, max)
  Min is now added as edgecase for Arb.bigDecimal(min, max)

myuwono commented on kotest-property/src/jvmMain/kotlin/io/kotest/property/arbitrary/bigdecimal.kt:
```diff
@@ -28,8 +28,11 @@ fun Arb.Companion.bigDecimal(scale: Int, roundingMode: RoundingMode) =
+      .plus(min)
+
+   return arbitrary(boundedEdgecases) {
       min.add(BigDecimal(Math.random()).multiply(max.subtract(min)))
```
@ciffer thanks heaps! do you mind also fixing this bug? It seems like currently this arb ignores the provided random source
```suggestion
      min.add(BigDecimal(it.random.nextDouble()).multiply(max.subtract(min)))
```

That is technically a different issue but would be great if you can also address that in this PR. Otherwise i'll raise a follow up PR for that. Thanks again!

ciffer committed:
BigDecimal arb can return edgecases outside min max limits #2834
  Edgecases for Arb.bigDecimal(min, max) are now always contrained to be in range [min, max)
  Min is now added as edgecase for Arb.bigDecimal(min, max)
